src/main/java/com/google/sps/data/MapsRequest.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.data;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.maps.errors.ApiException;
+import com.google.maps.GeoApiContext;
+import com.google.maps.GeoApiContext.Builder;
+import com.google.maps.GeocodingApi;
+import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.LatLng;
+
+/** 
+ * Class that create a GeoApiContext variable and handles requests to maps API.
+ */
+public class MapsRequest {
+  private static GeoApiContext geoApiContext = null;
+
+  /** 
+   * Returns the geoApiContext or creates one if it wasn't already created.
+   */
+  public static GeoApiContext getGeoApiContext() {
+    if (geoApiContext == null) {
+      geoApiContext = new GeoApiContext
+          .Builder()
+          .apiKey("")
+          .build();
+    }
+    return geoApiContext;
+  }
+
+  /**
+   * Makes a request to Geocoding API and returns the latitude and longitude associated with the
+   * address String.
+   */
+  public static LatLng getLocationFromAddress(String address) {
+    try {
+      GeocodingResult[] results =  GeocodingApi.geocode(getGeoApiContext(), address).await();
+      Gson gson = new GsonBuilder().setPrettyPrinting().create();
+      return results[0].geometry.location;
+    } catch(Exception e) {
```

Comment:
Why are you forcefully silencing exceptions here?

Let your caller handle the exceptions raised by the API. Don't make them handle `null`. Null does not contain an error message.

In general it's also a bad idea to handle `Exception`. This is akin to saying "I don't care about exception handling at any level shape or form". But you should! You'll generally want to handle more granular errors (e.g. IOException or whatever the API is documented to be returning).

Reply:
Thank you for the tip. I changed the function to throw the errors.
Is it okay for now to handle exceptions in NewRequestServlet.java in this manner:
```
catch (Exception e) {
        // Exception can be ApiException, IOException, InterruptedException
        response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
        return ;
      }
```

Reply:
Not really, you should not just blindly catch all exceptions. You should just enumerate the exact exceptions you want to catch and what you want to do in each individual case.

If you really want the same behavior, you can enumerate them: http://tutorials.jenkov.com/java-exception-handling/catching-multiple-exceptions.html. But that's not the case here:

ApiException seems fine to catch here and return back a 400 bad request (if you are sure the api request would succeed otherwise).
InterruptedException is when the server gets shut down from the commandline. You really shouldn't send a 400 bad request to the client (it's not a bad request, the issue here is you're shutting down the server).
IOException is when you can't connect to another server. Again nothing to do with the user's request in any way, and it's definitely not a bad request. This is a server issue connection issue.

For the last 2 I believe default servlet code will already handle and catch any exceptions, so you don't need to worry about it. You should only return a bad request, when you want to signal to the client that the request in its current form will NEVER succeed because it's simply invalid (e.g. latitude larger than 90.. it simply cannot be).

Generally only handle the exceptions that you are sure you want to handle. It's better to underhandle than to overhandle exceptions.